it/test/uk/gov/hmrc/tradergoodsprofiles/controllers/WithdrawAdviceControllerIntegrationSpec.scala

```diff
@@ -0,0 +1,298 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofiles.controllers
+
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock._
+import org.mockito.MockitoSugar.{reset, when}
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import play.api.Application
+import play.api.http.Status._
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.Json
+import play.api.libs.ws.WSClient
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+import uk.gov.hmrc.auth.core.AffinityGroup.Agent
+import uk.gov.hmrc.auth.core.{AuthConnector, Enrolment, InsufficientEnrolments}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.test.HttpClientV2Support
+import uk.gov.hmrc.tradergoodsprofiles.controllers.support.AuthTestSupport
+import uk.gov.hmrc.tradergoodsprofiles.services.UuidService
+import uk.gov.hmrc.tradergoodsprofiles.support.WireMockServerSpec
+
+import java.util.UUID
+import scala.concurrent.ExecutionContext
+
+class WithdrawAdviceControllerIntegrationSpec
+    extends PlaySpec
+    with GuiceOneServerPerSuite
+    with HttpClientV2Support
+    with AuthTestSupport
+    with WireMockServerSpec
+    with BeforeAndAfterEach
+    with BeforeAndAfterAll {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  private lazy val wsClient: WSClient = app.injector.instanceOf[WSClient]
+  private val recordId                = UUID.randomUUID().toString
+  private val uuidService             = mock[UuidService]
+  private val correlationId           = "d677693e-9981-4ee3-8574-654981ebe606"
+  private val withdrawReason          = "today"
+
+  private val url       = s"http://localhost:$port/$eoriNumber/records/$recordId/advice?withdrawReason=$withdrawReason"
+  private val routerUrl =
+    s"/trader-goods-profiles-router/traders/$eoriNumber/records/$recordId/advice?withdrawReason=$withdrawReason"
+
+  private val routerResponse = NO_CONTENT
+
+  override lazy val app: Application = {
+    wireMock.start()
+    configureFor(wireHost, wireMock.port())
+
+    GuiceApplicationBuilder()
+      .configure(configureServices)
+      .overrides(
+        bind[AuthConnector].to(authConnector),
+        bind[UuidService].to(uuidService),
+        bind[HttpClientV2].to(httpClientV2)
+      )
+      .build()
+  }
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    reset(authConnector)
+    stubRouterResponse(NO_CONTENT, routerResponse.toString)
```

Comment:
This can be replaced, you can then delete the variable at the top. 
```suggestion
    stubRouterResponse(NO_CONTENT, "204")
```